sips/fork_support.md

```diff
@@ -20,56 +20,247 @@ Client implementation should calculate when a new duty starts to decide if it st
+	byts, err := json.Marshal(f)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal ForkData")
+	}
+	ret := sha256.Sum256(byts)
+	return ret[:], nil
+}
+```
+
+### Structures access to types
+
+- `Share`: It's more appropriate for the structure to hold the information on the network that it's in, instead of the domain type.
+    Thus, _Share_ will contain a `NetworkID` field instead of a `DomainType` field.
+
+Regarding the indirect usage of Identifiers:
+- `Instance`:
+  - Must keep its identifier to send properly formed messages.
+- `Controller`:
+  - Should replace its _Identifier_ with an _IdentifierF_ getter function, passed by the Runner. With the function instead of the fixed value, it can compute the up-to-date identifier when needed.
+  - Since the controller's `config` must also be updated, the controller should also contain a _DomainTypeF_ function, instead of its fixed _DomainType_ field.
+
+```go
+
+type IdentifierF func() []byte // <-- new func type
+type DomainTypeF func() types.DomainType // <-- new func type
+
+type Controller struct {
+	IdentifierF IdentifierF // <-- replace Identifier []byte to Identifier getter function
+	Height     Height 
+	StoredInstances InstanceContainer
+	FutureMsgsContainer map[types.OperatorID]Height
+	DomainTypeF         DomainTypeF // <-- replace Domain types.DomainType to DomainType getter function
+	Share               *types.Share
+	config              IConfig
+}
+
+func (c *Controller) StartNewInstance(height Height, value []byte) error {
+	// ...
+
+	c.config.SetSignatureDomainType(c.DomainTypeF()) // <-- updates its config using its new DomainF
+    	
+	c.Height = height
+	
+	newInstance := c.addAndStoreNewInstance()
+	
+	newInstance.Start(value, height)
+
+	c.forceStopAllInstanceExceptCurrent()
+
+	return nil
+}
+
+func (c *Controller) addAndStoreNewInstance() *Instance {
+	i := NewInstance(c.GetConfig(), c.Share, c.IdentifierF(), c.Height) // <-- set instance's identifier as new up to date identifier
+
+	c.StoredInstances.addNewInstance(i)
+	
+	return i
 }
 
 ```
 
+An example of how the required functions for the new controller can be computed using the `BaseRunner` is shown below.
+
+```go
+type BaseRunner struct {
+	State          *State
+	Share          *types.Share
+	QBFTController *qbft.Controller
+	BeaconNetwork  types.BeaconNetwork
+	BeaconRoleType types.BeaconRole
+	highestDecidedSlot spec.Slot
+}
+
+func (b *BaseRunner) GetIdentifierF() func() []byte {
+	return func() []byte {
+		currentEpoch := b.BeaconNetwork.EstimatedCurrentEpoch()
+		domainType := b.Share.NetworkID.ForkAtEpoch(currentEpoch)
+		identifier := types.NewMsgID(domainType,b.Share.ValidatorPubKey[:],b.BeaconRoleType)
+		return identifier[:]
+	}
+}
+
+func (b *BaseRunner) GetDomainTypeF() func() types.DomainType {
+	return func() types.DomainType {
+		currentEpoch := b.BeaconNetwork.EstimatedCurrentEpoch()
+		return b.Share.NetworkID.ForkAtEpoch(currentEpoch)
```

Comment:
This returns DomainType?

Reply:
This returns a function that returns the current DomainType

Reply:
What I meant is that 		
```
return b.Share.NetworkID.ForkAtEpoch(currentEpoch)
```

will return forkdata and an error. It won't return the domain type.

Reply:
Oh sorry, yeah, updating it now